main.go

```diff
@@ -54,29 +55,100 @@ func main() {
 	if err := stepconf.Parse(&config); err != nil {
 		fail("Issue with input: %s", err)
 	}
+
 	if err := validateGoTemplate(config.PublicInstallPageMapFormat); err != nil {
 		fail("PublicInstallPageMapFormat - %s", err)
 	}
 
 	stepconf.Print(config)
-
 	fmt.Println()
-
 	log.SetEnableDebugLog(config.DebugMode)
 
 	absDeployPth, err := pathutil.AbsPath(config.DeployPath)
 	if err != nil {
 		fail("Failed to expand path: %s, error: %s", config.DeployPath, err)
 	}
 
-	filesToDeploy := []string{}
-
 	tmpDir, err := pathutil.NormalizedOSTempDirPath("__deploy-to-bitrise-io__")
 	if err != nil {
 		fail("Failed to create tmp dir, error: %s", err)
 	}
 
-	// Collect files to deploy
+	filesToDeploy := collectFilesToDeploy(absDeployPth, config, tmpDir)
```

Comment:
please try always use logging in the main thread. also please make sure that there is no code exit in the external functions, because the main thread expects an error return and handles the errors in the main thread.

Reply:
please separate housekeeping changes from the original task, as now it is really hard to see through what to check exactly.

Reply:
also when you exit at any function then it is impossible to follow the return/exit logic flow in the code

Reply:
Updated code: app only exits from main. Moved logging from methods to main, where it wa possible.